falcon/request.py

```diff
@@ -1028,6 +1059,22 @@ def _parse_form_urlencoded(self):
 
             self._params.update(extra_params)
 
+    def _parse_rfc_forwarded(self):
+        """rfc 7239"""
+        addr = []
+        for forwarded in self.env['HTTP_FORWARDED'].split(','):
+            for param in forwarded.split(';'):
+                param = param.strip().split('=', 1)
+                if len(param) == 1:
+                    continue
+                key, val = param
+                if key.lower() not in ('for', 'by'):
```

Comment:
It seems like we should only return "for" parameters to be consistent with the semantics of X-Forwarded-For, and to avoid ambiguity in the returned `addr` list. What do you think? As an alternative, we might compose a richer data structure that includes the "for" and "by" designations. To be honest, I don't have a good sense of how many apps would make use of the "by" parameter.